app/models/book.rb

```diff
@@ -1,2 +1,7 @@
 class Book < ApplicationRecord
+  has_many :stocks, class_name: "Shop::Stock", inverse_of: :book
+  has_many :shops, through: :stocks
+
+  has_many :not_empty_stocks, -> { where('value > 0') }, class_name: "Shop::Stock", inverse_of: :book
+  has_many :shops_with_book, through: :not_empty_stocks, class_name: 'Shop', source: :shop
```

Comment:
same question:
`book.stocks.not_empty`

Reply:
When you have additional conditional association it will be simpler to make queries. You don't need to think "hm, should I add scope for not empty stocks or not". You just use always not empty stocks or books.

Reply:
And as a bonus, you could do cool queries through ransack.